src/server.h

```diff
@@ -2409,6 +2419,27 @@ typedef struct {
     unsigned char *lpi; /* listpack iterator */
 } setTypeIterator;
 
+/* Structure for DB iterator that allows iterating across multiple slot specific dictionaries in cluster mode. */
+typedef struct dbIterator {
+    redisDb *db;
+    int slot;
+    int next_slot;
+    dictIterator di;
+    int dictType;
+} dbIterator;
```

Comment:
If we can, let's make this opaque and have APIs for accessors.

I also think this is the one usage of dictType that I think makes sense. I would prefer it be flags on the iterator type though, and not dictType.